common/log.c

```diff
@@ -490,6 +496,60 @@ internal_log_config_copy(struct log_config *dest, const struct log_config *src)
     }
 }
 
+bool_t
+internal_log_is_enabled_for_level(const enum logLevels log_level,
+                                  const bool_t override_destination_level)
+{
+    /* Is log initialized? */
+    if (g_staticLogConfig == NULL)
+    {
+        return 0;
+    }
+
+    /* Is there at least one log destination which will accept the message based on the log level? */
+    return (g_staticLogConfig->fd >= 0 
+                && (override_destination_level || log_level <= g_staticLogConfig->log_level))
+            || (g_staticLogConfig->enable_syslog 
+                && (override_destination_level || log_level <= g_staticLogConfig->syslog_level))
+            || (g_staticLogConfig->enable_console 
+                && (override_destination_level || log_level <= g_staticLogConfig->console_level));
+}
+
+bool_t
+internal_log_location_overrides_level(const char *function_name, 
+                          const char *file_name,
+                          const enum logLevels log_level)
+{
+    struct log_logger_level *logger = NULL;
+    int i;
+    
+    if (g_staticLogConfig == NULL)
+    {
+        return 0;
+    }
+    for (i = 0; i < g_staticLogConfig->per_logger_level->count; i++)
+    {
+        logger = (struct log_logger_level *)list_get_item(g_staticLogConfig->per_logger_level, i);
+
+        if ((logger->logger_type == LOG_TYPE_FILE
+                && 0 == g_strncmp(logger->logger_name, file_name, LOGGER_NAME_SIZE))
+            || (logger->logger_type == LOG_TYPE_FUNCTION
+                && 0 == g_strncmp(logger->logger_name, function_name, LOGGER_NAME_SIZE)))
+        {
+            if(logger->log_level < log_level)
+            {
+                return 0;
+            }
+            else
+            {
+                return 1;
+            }
```

Comment:
Wouldn't `return (log_level <= logger->log_level)` be a simpler way to express this?

Reply:
agreed, fixed